REGRESSION (273187@main): [ iOS17 Debug arm64 ] ASSERTION FAILED:bool WTF::RefCountedBase::hasOneRef() in http/tests/workers/service/postmessage-after-terminate.https.html result of a flaky crash

cdumez opened the pull request:
#### 0963c19c0ab8e2a0872897f1fb3399cc70948bcd
<pre>
REGRESSION (273187@main): [ iOS17 Debug arm64 ] ASSERTION FAILED:bool WTF::RefCountedBase::hasOneRef() in http/tests/workers/service/postmessage-after-terminate.https.html result of a flaky crash
<a href="https://bugs.webkit.org/show_bug.cgi?id=268029">https://bugs.webkit.org/show_bug.cgi?id=268029</a>
<a href="https://rdar.apple.com/121550392">rdar://121550392</a>

Unreviewed, very partial revert of 273187@main to address the assertion hit.

* LayoutTests/platform/ios-wk2/TestExpectations:
* Source/WebCore/workers/service/server/SWServerWorker.cpp:
(WebCore::SWServerWorker::contextConnection):

Canonical link: <a href="https://commits.webkit.org/273467@main">https://commits.webkit.org/273467@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8a5654786a287f5b9003ed9d2b71363544485aa2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35556 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14497 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37692 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38298 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/32045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36753 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16877 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11525 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30856 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36109 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12246 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/31650 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10755 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/10777 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31782 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39543 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32309 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32116 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/36737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10957 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8848 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/34796 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/12673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8115 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/11472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11736 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/workers/service/server/SWServerWorker.cpp:
```diff
@@ -179,7 +179,8 @@ const ClientOrigin& SWServerWorker::origin() const
 SWServerToContextConnection* SWServerWorker::contextConnection()
 {
-    return m_server ? protectedServer()->contextConnectionForRegistrableDomain(registrableDomain()) : nullptr;
+    // Unable to ref m_server here as it may have started destruction.
```
I wonder if this should be accessing a WeakPtr instead?

cdumez replied:
WeakPtr would not offer additional protection here.

brentfulgham reviewed (approved):
Looks good, but do we need some better mechanism to handle partially-destroyed objects in our smart-pointer plan? r=me

cdumez committed:
REGRESSION (273187@main): [ iOS17 Debug arm64 ] ASSERTION FAILED:bool WTF::RefCountedBase::hasOneRef() in http/tests/workers/service/postmessage-after-terminate.https.html result of a flaky crash
https://bugs.webkit.org/show_bug.cgi?id=268029
rdar://121550392

Unreviewed, very partial revert of 273187@main to address the assertion hit.

* LayoutTests/platform/ios-wk2/TestExpectations:
* Source/WebCore/workers/service/server/SWServerWorker.cpp:
(WebCore::SWServerWorker::contextConnection):

Canonical link: https://commits.webkit.org/273467@main
